File: 백준/Silver/7785. 회사에 있는 사람/회사에 있는 사람.cc
// baekjoon 7785

#include <bits/stdc++.h>
#define endl '\n'

using namespace std;

int main()
{
    int N;
    cin >> N;

    unordered_map<string, string> M;
    while (N--) {
        string str1, str2;
        cin >> str1 >> str2;
        M[str1] = str2;
    }

    vector<string> ans;
    for (auto [name, status] : M) {
        if (status == "enter")
            ans.push_back(name);
    }

    sort(ans.begin(), ans.end(), greater<string>());
    for (auto e : ans)
        cout << e << endl;
}

File: 백준/Silver/7785. 회사에 있는 사람/회사에 있는 사람.js
/**
 * Input
 */
const filePath = process.platform === "linux" ? "/dev/stdin" : "./src/input.txt";
const input = require("fs").readFileSync(filePath).toString().trim().split("\n");

/**
 * Solution
 */
const people = {};
const ans = [];
input.slice(1).forEach(e => {
    const [name, status] = e.trim().split(' ').map(e => e.trim());
    people[name] = status;
});

for (const [name, status] of Object.entries(people)) {
    if (status === "enter")
        ans.push(name);
}

console.log(ans.sort().reverse().join('\n'));